sherpa/plot/pylab_backend.py

```diff
@@ -19,857 +19,822 @@
-    if not overplot and clearwindow:
-        clear_window()
-
-    return plt.gca()
-
-
-def setup_plot(axes, title, xlabel, ylabel, xlog=False, ylog=False):
-    """Basic plot setup.
-
-    Parameters
-    ----------
-    axes
-        The plot axes (output of setup_axes).
-    title, xlabel, ylabel : str or None
-        The plot, x-axis, and y-axis titles. They are skipped if
-        the empty string or None.
-    xlog , ylog : bool
-        Should the scale be logarithmic (True) or linear (False)?
-
-    """
-
-    axes.set_xscale('log' if xlog else 'linear')
-    axes.set_yscale('log' if ylog else 'linear')
-
-    if title:
+kwargs_doc = ChainMap(updated_kwarg_docs, orig_kwargs_doc)
+
+
+class PylabBackend(BasicBackend):
+
+    name = 'pylab'
+    '''Alternative string names for plotting backend.
+
+    This differs from the class name, because external code such
+    as ciao_contrib scripts might depend on this name being present.
+    '''
+
+    translate_dict = {'linestyle': {
+        'noline': ' ',
+        'solid': '-',
+        'dot': ':',
+        'dash': '--',
+        'dotdash': '-.',
+        'None': ' ', },
+        'label': {None: '_nolegend_'},
+    }
+    '''Dict of keyword arguments that need to be translated for this backend.
+
+    The keys in this dict are keyword arguments(e.g. ``'markerfacecolor'``)
+    and the values are one of the following:
+    - A dict where the keys are the backend-independent values and the values
+    are the values expressed for this backend. For values not listed in the
+    dict, no translation is done.
+    - A callable. The callable translation function is called with any argument
+    given and allows the backend arbitrary translations. It should, at the
+    very least, accept all backend independent values for this parameter
+    without error.
+
+    Example:
+
+       >>> translate_dict = {'markerfacecolor': {'k': (0., 0., 0.)},
+       ...                   'alpha': lambda a: 256 * a}
+
+    This translates the color 'k' to tuple of RGB values and alpha values
+    to a number between 0 and 256.
+    '''
+
+    def colorlist(self, n):
+        '''Generate the list of n colors for use in multi-line plots.
+
+        Generally, the color will be ordered in some way and do not repeat or
+        do so only after a large number of colors.
+        Different backends might generate different lists of colors.
+
+        Parameters
+        ----------
+        n : int
+            Number of colors requested
+
+        Returns
+        -------
+        colors : list
+            list of color specifiers
+        '''
+        return plt.cm.inferno(numpy.linspace(0, 1, n))
+
+    def clear_window(self):
+        """Clear default pyplot figure."""
+        plt.clf()
+
+    def __exit__(self, exec_type, value, traceback):
+        self.set_window_redraw(True)
+        if plt.isinteractive():
+            plt.draw()
+        return False
+
+    def set_window_redraw(self, redraw):
```

Comment:
I think we can just delete this method.

Reply:
I wish I was sure that that's true - I won't do it in this PR, but look more closely in my editor. Even if we don't do it here, there is some more clean-up of the plotting backend definitions coming when I get to the datastack plots and some of the other things we discussed in e.g. #1868 or #1887, so we don't have to do that right now if it's not obvious.

Reply:
I looked. We can; I'm removing it.